pallets/did/src/lib.rs

```diff
@@ -213,7 +213,10 @@ pub mod pallet {
 			+ DeriveDidCallAuthorizationVerificationKeyRelationship;
 
 		/// Type for a DID subject identifier.
-		type DidIdentifier: Parameter + DidVerifiableIdentifier + MaxEncodedLen;
+		type DidIdentifier: Parameter
+			+ DidVerifiableIdentifier<AccountIdOf<Self>>
+			+ MaxEncodedLen
+			+ From<AccountIdOf<Self>>;
```

Comment:
This cannot be removed since it's required by `create_from_account`. There we derive the did_identifier from the calling account and set an unrelated key as the authentication key.

DidIdentifier and AccountId can still be completely different types. There just have to be a way to derive a DidIdentifier from an AccountId.

Reply:
This goes back to the discussion we had last time. Perhaps the `create_from_account` extrinsic should only exist for instances of the pallet where the DID identifier can indeed be derived from the account ID. Otherwise, this is a strong assumption we are introducing here. I mean, we can always relax this requirement if we need to in the future.

Reply:
Yes and as I stated the last time, we can't add where clauses to calls. The macro derives an enum and some functions from this call and the where clause doesn't translate well there. There is no way to add the requirement to all the generated code. So it has to be on the whole pallet.

I could try to setup some conditional compilation and a feature flag or something. But I don't think this is worth the trouble.

```
  error[E0277]: the trait bound `<T as pallet::Config>::DidIdentifier: From<<T as frame_system::Config>::AccountId>` is not satisfied
      --> /pallets/did/src/lib.rs:510:12
       |
  510  |       #[pallet::call]
       |  _______________^
  511  | |     impl<T: Config> Pallet<T>
  512  | |     where
  513  | |         T::AccountId: AsRef<[u8; 32]> + From<[u8; 32]>,
  ...    |
  1211 | |         #[pallet::weight(<T as pallet::Config>::WeightInfo::create_from_account())]
  1212 | |         pub fn create_from_account(
       | |__________________________________^ the trait `From<<T as frame_system::Config>::AccountId>` is not implemented for `<T as pallet::Config>::DidIdentifier`
       |
  note: required by a bound in `pallet::Pallet::<T>::create_from_account`
      --> /pallets/did/src/lib.rs:1217:22
       |
  1212 |         pub fn create_from_account(
       |                ------------------- required by a bound in this associated function
  ...
  1217 |             T::DidIdentifier: From<AccountIdOf<T>>,
       |                               ^^^^^^^^^^^^^^^^^^^^ required by this bound in `Pallet::<T>::create_from_account`
  help: consider further restricting the associated type
       |
  510  |     #[pallet::call where <T as pallet::Config>::DidIdentifier: From<<T as frame_system::Config>::AccountId>]
       |                    ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

  For more information about this error, try `rustc --explain E0277`.
  error: could not compile `did` (lib) due to previous error
```

Reply:
Yes, we will think about it in the future if and when the time comes.